numpyro/contrib/render.py

```diff
@@ -43,31 +46,43 @@ def model(data):
     :param callable model: A model to inspect.
     :param model_args: Optional tuple of model args.
     :param model_kwargs: Optional dict of model kwargs.
-    :param int num_tries: Optional number times to trace model to detect
-        discrete -> continuous dependency.
     :rtype: dict
     """
     model_args = model_args or ()
     model_kwargs = model_kwargs or {}
 
-    trace = handlers.trace(handlers.seed(model, 0)).get_trace(
-        *model_args, **model_kwargs
-    )
-    obs_sites = [
-        name
-        for name, site in trace.items()
-        if site["type"] == "sample" and site["is_observed"]
-    ]
-
     def _get_dist_name(fn):
         if isinstance(
             fn, (dist.Independent, dist.ExpandedDistribution, dist.MaskedDistribution)
         ):
             return _get_dist_name(fn.base_dist)
         return type(fn).__name__
 
+    def get_trace():
+        # We use `init_to_sample` to get around ImproperUniform distribution,
+        # which does not have `sample` method.
+        subs_model = handlers.substitute(
+            handlers.seed(model, 0),
+            substitute_fn=init_to_sample,
+        )
+        trace = handlers.trace(subs_model).get_trace(*model_args, **model_kwargs)
```

Comment:
In Pyro we [use an effect handler](https://github.com/pyro-ppl/pyro/blob/b134ae8ec2380ca86046afcff28b337a0bac86c9/pyro/infer/inspect.py#L48) to replace the provenance of each sample value by the singleton `site["name"]`, thereby tracking only direct dependencies rather than transitive dependencies.  Can you point me to where in this logic you replace transitive dependencies by the new sample site name?